### DISCUSSION TITLE
Feature Request: use Happy Eyeballs algorithm instead of IPv4-first

### DISCUSSION THREAD

**Discussion by basisbit** [OvenMediaLabs/OvenMediaEngine#1383] [Ideas] [answered]
Currently OME prioritizes IPv4 over IPv6 when a hostname can be resolved to both. This works well for most cases, but not for all (for example when a system has A and AAAA record but the connection can only be established (or faster) over IPv6.

The IETF suggests to use the Happy Eyeballs algorithm instead for a faster / more successful connection setup.

A summary can be found at https://en.wikipedia.org/wiki/Happy_Eyeballs

**Comment by dimiden** [comment, marked as answer]
Thank you for the valuable information!
When I was working on IPv6 development, I had to ponder over which one to choose between IPv4 and IPv6, and it seems like RFC8305 could provide an answer to that.
However, I don't have the time to delve into it right now, so I'll examine whether this algorithm can be applied to OME when I have the time in the future.
